chore(http): Refresh headers ~[s->x] (#36862)
mdn/content: files/en-us/web/http/headers/sourcemap/index.md

{{HTTPSidebar}}

> The HTTP **`SourceMap`** {{Glossary("response header")}} links generated code to a {{Glossary("source map")}}, enabling the browser to reconstruct the original source and present the reconstructed original in the debugger.

Reviewer: This mixes what a source map does and what HTTP is doing.

```suggestion
The HTTP **`SourceMap`** {{Glossary("response header")}} provides the location of a {{Glossary("source map")}} for the resource.
```

We could explain what a source map does here too, but I think that it is better that they go to the glossary link if they don't know. We can also explain thi sin the examples.

Author: Good point. See f20875ebd209e00620978ab092d3a6ce6662e9ec

> The HTTP `SourceMap` header has precedence over a source annotation (`sourceMappingURL=path-to-map.js.map`), and if both are present, the header URL is used to resolve the source map file.

Reviewer: Link to `sourceMappingURL`?

Author: I don't think we have anything good internally yet. At least I don't find anything

Reviewer: Cool. Perhaps then we should at least link to the document where it _should be_?

### Directives

- `<url>`
  - : A relative (to the request URL) or absolute URL pointing to a source map file.

## Examples

### Linking to a source map using `SourceMap`

> The following response contains an absolute

Author: TODO

### Directives

- `<url>`
  - : A relative (to the request URL) or absolute URL pointing to a source map file.

## Examples

### Linking to a source map using the `SourceMap` header

The following response contains an absolute path in the `SourceMap` header.

```http
HTTP/1.1 200 OK
Content-Type: application/javascript
SourceMap: /path/to/file.js.map

<optimized-javascript>
```

> Developer tools use the mappings between optimized code and the original source to improve readability, which can make debugging easier.

Reviewer: ```suggestion
Developer tools use the source map to reconstruct the original source from the optimized JavaScript returned in the response, allowing developers to debug the original code rather than the format that has been optimized for sending.
```
